packages/clerk-js/src/core/clerk.ts

```diff
@@ -1345,4 +1347,28 @@ export default class Clerk implements ClerkInterface {
       throw new Error('ClerkJS components are not ready yet.');
     }
   }
+
+  shouldRedirectFAPIInitiatedFlow = async (): Promise<boolean> => {
+    const redirectUrl = new URLSearchParams(window.location.search).get('redirect_url');
+    if (
+      this.instanceType !== 'production' &&
+      redirectUrl !== null &&
+      isRedirectForFAPIInitiatedFlow(this.frontendApi, redirectUrl)
+    ) {
+      // don't redirect if user is not signed in and referrer is sign in url
+      if (
+        !this.session &&
+        this.#environment &&
+        window.location.href.startsWith(this.#environment.displayConfig.signInUrl)
+      ) {
+        return false;
+      }
+
+      const buildUrlWithAuthParams: BuildUrlWithAuthParams = { useQueryParam: true };
+      await this.navigate(this.buildUrlWithAuth(redirectUrl, buildUrlWithAuthParams));
+      return true;
+    }
+
+    return false;
```

Comment:
🙃 i would suggest to use guards instead of big control flow blocks. eg
```typescript
    const prodInstance =  this.instanceType === 'production'
    const shouldRedirect =  redirectUrl !== null && isRedirectForFAPIInitiatedFlow(this.frontendApi, redirectUrl);
    
    if (!prodInstance && shouldRedirect) {
      return false
    }
   
     const userSignedIn = this.session;
     const signInUrl = this.#environment?.displayConfig?.signInUrl
     const comesFromSignInUrl = signInUrl && window.location.href.startsWith(signInUrl);
    // don't redirect if user is not signed in and referrer is sign in url
    if (!userSignedIn && comesFromSignInUrl) {
      return false;
    }
    // .... redirect code 
    return true;
```